### ISSUE TITLE
missing docs: rank_genes_groups expecting log-transformed data

### ISSUE THREAD

**Issue by redst4r** [scverse/scanpy#967]
Hi,

not really a bug, more of a documentation issue: 
`sc.tl.rank_genes_groups` seems to expect log-transformed data (be it in `.X` or `.raw.X`).
To my knowledge this is **not mentioned in the docs**. I just ran into trouble and then found out via #671 and #517 . 

It's not a problem for the p-values (if the data is not log-transformed it just does the t-test etc on the counts), but the resulting fold-changes are wrong (it essentially tries to undo the expected log-transform):
```python
foldchanges = (np.expm1(mean_group) + 1e-9) / (np.expm1(mean_rest) + 1e-9)
```
I was totally unaware of this (been using scanpy for quite a while), especially since I usually store the plain raw counts in the `adata.raw` field, which is used *by default* in `rank_genes_groups`.

We should at least mention it in the docstring, but these things are easy to overlook too...
Is there any way that scanpy records the transformations you've done to the data (and if the log is missing, just spits out an error in rank_genes_groups)?

Also, is there a "best-practice" guide of what data to store in which parts of the AnnData objects?
`.raw` and the `.layer` dont have a lot of documentation

#### Versions:

> scanpy==1.4.4.post1 anndata==0.6.22.post1 umap==0.3.10 numpy==1.17.3 scipy==1.3.1 pandas==0.25.2 scikit-learn==0.21.3 statsmodels==0.10.1 python-igraph==0.7.1 louvain==0.6.1

**Comment by giovp** [comment]
> I was totally unaware of this (been using scanpy for quite a while), especially since I usually store the plain raw counts in the adata.raw field, which is used by default in rank_genes_groups.

I think this is what most people do. 

I feel this could be a bug:  having `raw=True` by default coupled with this: 

```python
foldchanges = (np.expm1(mean_group) + 1e-9) / (np.expm1(mean_rest) + 1e-9)
```

@Koncopd what's your take?

**Comment by giovp** [comment]
@Koncopd kind reminders. I can quickly fix the docs if it's desired behaviour.

**Comment by Koncopd** [comment]
@giovp sorry missed this, i'll check.

**Comment by Koncopd** [comment]
@giovp 
In the docs it is mentioned that it expects logarithmized data.
As for `raw=True` i am not sure it is a problem. For example, in [the tutorial](https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html) normalized and logarithmized data is saved to `raw` to be used further in `rank_genes_groups`. Do you think it is a problem?

I don't actually consider raw as a container only for raw counts, i think it is a matter of filtered genes vs unfiltered and so on.

**Comment by giovp** [comment]
ok yes I get it now. Still if they are integers, than this doesn't make sense right?
`foldchanges = (np.expm1(mean_group) + 1e-9) / (np.expm1(mean_rest) + 1e-9)`.

I think it makes sense to mention it in the docstrings and also use the check_nonnegative_integers from _utils as it's done in highvar genes seurat_v3. 
What do you think?

**Comment by Koncopd** [comment]
@giovp it is already in docstring, i have added the check also.

**Comment by giovp** [comment]
amazing, thank you @Koncopd ! @redst4r I think this should be fixed, the docs explicitly mention this and now there is a check for integers values, closed with #1750

**Comment by alexlenail** [comment]
This behavior was not what I expected: 

https://github.com/scverse/scanpy/blob/2e98705347ea484c36caa9ba10de1987b09081bf/scanpy/tools/_rank_genes_groups.py#L536-L537

What is the motivation to use raw by default if it exists? Shouldn't that be documented here? https://scanpy.readthedocs.io/en/stable/generated/scanpy.tl.rank_genes_groups.html
